server/player/player.go

```diff
@@ -623,6 +637,25 @@ func (p *Player) Hurt(dmg float64, src world.DamageSource) (float64, bool) {
 	return totalDamage, true
 }
 
+// applyTotemEffects is an unexported function that is used to handle totem effects.
+func (p *Player) applyTotemEffects() {
+	p.addHealth(1 - p.Health())
+
+	for _, e := range p.Effects() {
+		p.RemoveEffect(e.Type())
+	}
+
+	p.AddEffect(effect.New(effect.Regeneration{}, 2, time.Second*40))
+	p.AddEffect(effect.New(effect.FireResistance{}, 1, time.Second*40))
+	p.AddEffect(effect.New(effect.Absorption{}, 1, time.Second*5))
```

Comment:
This is meant to give absorption 2 as per https://minecraft.wiki/w/Totem_of_Undying#Effect
> then grants 40‌[[BE](https://minecraft.wiki/w/Bedrock_Edition) only] or 45‌[[JE](https://minecraft.wiki/w/Java_Edition) only] seconds[[2]](https://minecraft.wiki/w/Totem_of_Undying#cite_note-2) of [Regeneration](https://minecraft.wiki/w/Regeneration) II, 40 seconds of [Fire Resistance](https://minecraft.wiki/w/Fire_Resistance) I and 5 seconds of [Absorption](https://minecraft.wiki/w/Absorption) II